pynamodb_attributes/timestamp.py

```diff
@@ -25,10 +25,14 @@ def serialize(self, value: datetime) -> str:
         return str(int(value.timestamp() * self._multiplier))
 
     def __set__(self, instance: Any, value: Optional[Any]) -> None:
```

Comment:
PynamoDB already validates nulls during serialization. If we want it to happen earlier (on assignment), we should probably change `Attribute.__set__` rather than do it in every attribute.

Can we instead wrap with `if value is not None:`, e.g.
```python
if value is not None:
    if not isinstance(value, datetime):
        raise TypeError(f"value has invalid type '{type(value)}'; datetime expected")
    if value.tzinfo is None or value.tzinfo.utcoffset(value) is None:
        raise TypeError("aware datetime expected")

return super().__set__(instance, value)
```